docs/learning/transformer/branching-imodels.md

```diff
@@ -0,0 +1,185 @@
+# iModel Transformation and iModel Branching
+
+## What is branching
+
+It can be useful in some applications to take iModels and their linear concept of history, and introduce the concept of
+*branching*, where an iModel diverges into two separate iModels after a changeset, allowing each to have new distinct changes and
+a distinct history. Typically, one of them is referred to as the *master iModel*, and the other the *branch iModel*,
+with the master iModel being the one that existed before the branch was created from it.
+Importantly, these branches can often be used to updates to one branch iModel while the master iModel
+can received updates from other contexts. Usually the intention is to move the changes in the branch iModel's history back into
+the master iModel. It can also be useful to update a branch with changes that the master received since their divergence.
+
+## Provenance
+
+*Provenance* is the tracking of where an entity originated from in an iModel. All connectors
+store provenance as they connect entities from some source format into an iModel. The [IModelTransformer]($transformer) will
+also store provenance as it makes changes to a target; provenance will be used during synchronizations to identify which element
+in the branch came from the master in order to merge and append changes.
+
+The [IModelTransformer]($transformer) has several options defined in the [IModelTransformOptions]($transformer) type
+that configure what provenance is stored, and particular configurations are required for performing each branching operation
+with the transformer.
+
+Provenance is stored for entities by adding an [ExternalSourceAspect]($backend) to them, which contains properties referencing the original
+repository they came from. The ExternalSourceAspects must reference an element that represents the real-world external source from which
+their corresponding entity was derived; in the case of transformation where the external source is an external iModel, the iModel is best
+conventionally represented with an [ExternalSource]($backend) element in a [RepositoryLink]($backend) element.
+
```

Comment:
yeah element Ids are not *supposed* to be the way to correlate anything. There is a transformer option for preserving them when just filtering because that makes it easier to use some existing display system code to compare an iModel with its branch that has the same Ids. But I'd prefer to think of that as an implementation detail of the display system that it only supports id-preserved branches. I will make note of it and federated ids

Reply:
@abruzas could you take a look at this and other unresolved comments again and see if you have any more feedback?

Reply:
@MichaelBelousov content-wise, looks good now (at least I can't spot anything missing). 

Though, I haven't tested the references (not sure what's the procedure for that, probably to build the docs locally?).

Reply:
I think I can resolve them for you... you should be able to, however

Reply:
That makes sense. I'll resolve them then.

Reply:
@abruzas go [here](https://github.com/orgs/iTwin/teams/devs/members) and request to join.